### ISSUE TITLE
How does the option Selection work?

### ISSUE THREAD

**Issue by owos** [chujiezheng/LLM-MCQ-Bias#6]
https://github.com/chujiezheng/LLM-MCQ-Bias/blob/e78748b673346f3b307e728e68eb48a0f1baf2bc/code/eval_clm_utils.py#L191

Hi there, thank you for your repo and code.
I am currently trying to replace some of this work on another data and I'm stuck at this point.
I keep getting the wrong option `A` for every sample I pass through the my model.
I currently do not understand what is happening here and why it works.

```
input_ids = input_ids[..., -1536:]
        with torch.no_grad():
            logits = model(
                input_ids=input_ids,
            ).logits[:, -1].view(-1)

        option_indices = [toker(f': {e}').input_ids[-1] for e in option_ids] + \
            [toker(f':{e}').input_ids[-1] for e in option_ids]
        probs = F.softmax(
            logits[..., option_indices], dim=-1
        ).detach().cpu().to(torch.float32).numpy()
        probs = probs.reshape(2, len(option_ids)).sum(axis=0)
        sampled = option_ids[np.argmax(probs)]
```

**Comment by chujiezheng** [comment]
The code contains the following steps:
1. Truncate input_ids to avoid out-of-memory;
2. Feed the input_ids into the model and obtain the prediction logits at the last position;
3. Since `A` and ` A` can be two different tokens, we gather the probabilities of the token with prefix space and the one without prefix space (from the line `option_indices = ...` to the line `probs = probs.reshape(2, len(option_ids)).sum(axis=0)`;
4. Obtain the final probability distribution and the argmax one.
